kotlin/samples/hello-terminal/terminal-workflow/src/main/java/com/squareup/sample/helloterminal/terminalworkflow/ShareOperator.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sample.helloterminal.terminalworkflow
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.broadcastIn
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.coroutines.EmptyCoroutineContext
+
+/**
+ * Shares the source flow between all downstream collectors.
+ * The flow is collected when the first downstream collector starts, and cancelled after the last
+ * downstream collector is cancelled. To control the dispatcher or other context elements of the
+ * coroutine on which this flow is collected, use the `flowOn` operator before calling `share`.
+ * To control buffering, use the `buffer` operator before calling `share`.
+ *
+ * Should be removed once the
+ * [official share operator](https://github.com/Kotlin/kotlinx.coroutines/issues/1261) lands.
+ */
+@UseExperimental(ExperimentalCoroutinesApi::class, FlowPreview::class)
+internal fun <T> Flow<T>.share(): Flow<T> {
```

Comment:
Unfortunately, I had to write a custom `share` operator for this, since the keystroke worker is used by two separate workflows simultaneously, and without it they each get alternating keystrokes. There is an issue open on the coroutines project to implement a standard operator (https://github.com/Kotlin/kotlinx.coroutines/issues/1261).

Reply:
Ray filed as #454